ui-v2/src/hooks/use-favicon.ts

```diff
@@ -0,0 +1,66 @@
+import { useEffect } from "react";
+
+type StateType =
+	| "SCHEDULED"
+	| "PENDING"
+	| "RUNNING"
+	| "COMPLETED"
+	| "FAILED"
+	| "CANCELLED"
+	| "CANCELLING"
+	| "CRASHED"
+	| "PAUSED";
+
+function getPreferredColorScheme(): "dark" | "light" | "no-preference" {
+	if (window.matchMedia("(prefers-color-scheme: dark)").matches) {
+		return "dark";
+	}
+	if (window.matchMedia("(prefers-color-scheme: light)").matches) {
+		return "light";
+	}
+	return "no-preference";
+}
+
+/**
+ * A hook that sets the browser favicon based on the provided state type.
+ * Resets the favicon to the default when the component unmounts.
+ *
+ * @param stateType - The state type to display in the favicon (e.g., "COMPLETED", "FAILED")
+ * @returns void
+ *
+ * @example
+ * ```tsx
+ * // Set favicon based on task run state
+ * useFavicon(taskRun.state_type);
+ * ```
+ */
+export function useFavicon(stateType: StateType | null | undefined): void {
```

Comment:
The `useFavicon` name is too general. Update the name to communicate that it only handles updating the favicon to a state icon.